BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
++    private Component permissionMessage;
++    // Waterfall end
+ 
+     /**
+      * Construct a new command with no permissions or aliases.
+@@ -66,4 +71,24 @@ public abstract class Command
+     {
+         return permission == null || permission.isEmpty() || sender.hasPermission( permission );
+     }
++
++    // Waterfall start - add previously Lombok-generated getters/setters - we're going on an adventure!
++    public String getPermissionMessage() {
++        return LegacyComponentSerializer.legacySection().serialize(permissionMessage);
++    }
++
++    protected void setPermissionMessage(final String permissionMessage) {
++        this.permissionMessage = LegacyComponentSerializer.legacySection().deserialize(permissionMessage);
++    }
++    // Waterfall end
++
++    // Waterfall start - we're going on an adventure!
++    public Component permissionMessage() {
++        return permissionMessage;
++    }
++
++    protected void permissionMessage(final Component permissionMessage) {
++        this.permissionMessage = permissionMessage;
++    }
++    // Waterfall end
+ }
+diff --git a/api/src/main/java/net/md_5/bungee/api/score/Objective.java b/api/src/main/java/net/md_5/bungee/api/score/Objective.java
+index 0dd25646..f96ef1e4 100644
+--- a/api/src/main/java/net/md_5/bungee/api/score/Objective.java
++++ b/api/src/main/java/net/md_5/bungee/api/score/Objective.java
+@@ -2,6 +2,7 @@ package net.md_5.bungee.api.score;
+ 
+ import lombok.AllArgsConstructor;
+ import lombok.Data;
++import net.kyori.adventure.text.Component; // Waterfall - we're going on an adventure!
+ 
+ /**
+  * Represents an objective entry.
+@@ -15,6 +16,12 @@ public class Objective
+      * Name of the objective.
+      */
+     private final String name;
++    // Waterfall start - we're going on an adventure!
++    /**
++     * Display name of the objective.
++     */
++    private final Component displayName;
++    // Waterfall end
+     /**
+      * Value of the objective.
+      */
+@@ -23,4 +30,10 @@ public class Objective
+      * Type; integer or hearts
+      */
+     private String type;
++
++    // Waterfall start - write previously Lombok-generated constructor - we're going on an adventure!
++    public Objective(final String name, final String value, final String type) {
++        this(name, Component.empty(), value, type);
++    }
++    // Waterfall end
+ }
+diff --git a/api/src/main/java/net/md_5/bungee/api/score/Team.java b/api/src/main/java/net/md_5/bungee/api/score/Team.java
+index 849ba1cf..db3f95b0 100644
+--- a/api/src/main/java/net/md_5/bungee/api/score/Team.java
++++ b/api/src/main/java/net/md_5/bungee/api/score/Team.java
+@@ -6,6 +6,10 @@ import java.util.HashSet;
+ import java.util.Set;
+ import lombok.Data;
+ import lombok.NonNull;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.text.Component;
++import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
++// Waterfall end
```

Comment:
see above

Reply:
Got nothing but :thonking: to say to this, since those are in the right place